### ISSUE TITLE
Pulling up a diff using Diffview raises an error

### ISSUE THREAD

**Issue by dbatten5** [NeogitOrg/neogit#1055]
### Description

When I run `d` to pull up a diff view using the `Diffview` integration, it raises an error:
`[Diffview.nvim] Failed to create diff buffer: ':0:{filename}'`. The view is like the following
![image](https://github.com/NeogitOrg/neogit/assets/17271241/578b09fb-fe90-468b-847d-89408e9116d7)

I recognise this is a `Diffview` error, however if I pull up a diff view manually using `DiffviewOpen`, I get the correct diff:
![image](https://github.com/NeogitOrg/neogit/assets/17271241/3a5c1d52-9c19-4b12-8628-92922f1cb8c2)

### Neovim version

NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1700008891

### Operating system and version

macOS 14.2

### Steps to reproduce

1. Open nvim
2. Run `:Neogit` (with `Diffview` integration)
3. Press `d` on a modified file

### Expected behavior

An error occurs and the diff looks like it's against an empty file

### Actual behavior

The correct diff should be displayed.

Here's the output of `DiffviewLog`
```
[INFO  2023-12-21 11:40:19.043 +0000] .../diffview.nvim/lua/diffview/api/views/diff/diff_view.lua:35: [api] Creating a new Custom DiffView.
[ERROR 2023-12-21 11:40:19.102 +0000] ...re/nvim/lazy/diffview.nvim/lua/diffview/scene/window.lua:106: ...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:187: The coroutine failed with this message: 
	context: cur_thread=<thread 0x01009c6888> co_thread=<thread 0x0100d6fbd0> co_func=.../share/nvim/lazy/diffview.nvim/lua/diffview/vcs/file.lua:199
...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:187: The coroutine failed with this message: 
	context: cur_thread=<thread 0x0100d6fbd0> co_thread=<thread 0x0100feac70> co_func=.../share/nvim/lazy/diffview.nvim/lua/diffview/vcs/file.lua:180
...re/nvim/lazy/neogit/lua/neogit/integrations/diffview.lua:20: Git show did not end with a blankline
stack traceback:
	[C]: in function 'error'
	...re/nvim/lazy/neogit/lua/neogit/integrations/diffview.lua:20: in function 'remove_trailing_blankline'
	...re/nvim/lazy/neogit/lua/neogit/integrations/diffview.lua:92: in function 'get_data'
	.../share/nvim/lazy/diffview.nvim/lua/diffview/vcs/file.lua:183: in function <.../share/nvim/lazy/diffview.nvim/lua/diffview/vcs/file.lua:180>
	[C]: in function 'xpcall'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:361: in function <...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:358>
stack traceback:
	[C]: in function 'error'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:369: in function <...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:358>
stack traceback:
	[C]: in function 'error'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:187: in function 'raise'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:215: in function 'step'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:412: in function 'produce_data'
	.../share/nvim/lazy/diffview.nvim/lua/diffview/vcs/file.lua:251: in function <.../share/nvim/lazy/diffview.nvim/lua/diffview/vcs/file.lua:199>
	[C]: in function 'xpcall'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:361: in function <...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:358>
stack traceback:
	[C]: in function 'error'
	...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:369: in function <...cal/share/nvim/lazy/diffview.nvim/lua/diffview/async.lua:358>
[ERROR 2023-12-21 11:40:19.102 +0000] ...cal/share/nvim/lazy/diffview.nvim/lua/diffview/utils.lua:83: Failed to create diff buffer: ':0:hello.py'
```

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({
  integrations = {
    telescope = true,
    diffview = true,
  },
  kind = 'split',
}) -- For instance, setup Neogit
```

**Comment by swarn** [comment]
Possibly the same as #1024?

**Comment by CKolkey** [comment]
Thanks for the reports :)
